src/linker/Linker/Annotations.cs

```diff
@@ -192,9 +196,20 @@ public void Mark (CustomAttribute attribute, in DependencyInfo reason)
 			Tracer.AddDirectDependency (attribute, reason, marked: true);
 		}
 
+		public IEnumerable<IMetadataTokenProvider> MarkedPending ()
```

Comment:
This sounds like it actually performs the marking. I would rename to at least `GetMarkedPending`...